controllers/gitrepository_controller.go

```diff
@@ -744,9 +745,10 @@ func (r *GitRepositoryReconciler) gitCheckout(ctx context.Context,
 			authOpts.TransportOptionsURL = fmt.Sprintf("ssh://%s/%s/%d", obj.Name, obj.UID, obj.Generation)
 		} else {
 			e := &serror.Stalling{
-				Err:    fmt.Errorf("git repository URL has invalid transport type: '%s'", obj.Spec.URL),
+				Err:    fmt.Errorf("git repository URL '%s' has invalid transport type, expected to be one of http, https, ssh", obj.Spec.URL),
```

Comment:
Maybe we should say "supported protocols", instead of expected?

Reply:
Currently the CLI does something similar:

`✗ git URL scheme 'stl' not supported, can be: ssh, http and https`